tx-indexer/src/collector.rs

```diff
@@ -109,23 +111,92 @@ async fn save_outcomes_and_receipts(
     Ok(())
 }
 
-async fn save_outcome_and_receipt_to_shard(
+#[cfg_attr(feature = "tracing-instrumentation", tracing::instrument(skip_all))]
+async fn save_receipts_and_outcomes_details(
     db_manager: &std::sync::Arc<Box<dyn database::TxIndexerDbManager + Sync + Send + 'static>>,
+    tx_collecting_storage: &std::sync::Arc<crate::storage::CacheStorage>,
     shard_id: database::primitives::ShardId,
     receipts: Vec<readnode_primitives::ReceiptRecord>,
     outcomes: Vec<readnode_primitives::OutcomeRecord>,
 ) {
+    match save_outcome_and_receipt_to_shard(
+        db_manager,
+        shard_id,
+        receipts.clone(),
+        outcomes.clone(),
+    )
+    .await
+    {
+        Ok(_) => {
+            for receipt in receipts {
+                tx_collecting_storage
+                    .remove_receipt_from_watching_list(&receipt.receipt_id.to_string())
+                    .await
+                    .map_err(|err| {
+                        tracing::error!(
+                            target: crate::INDEXER,
+                            "Failed to remove receipt from watching list {}: Error {}",
+                            receipt.receipt_id.to_string(),
+                            err
+                        );
+                        err
+                    })
+                    .ok();
+            }
+        }
+        Err(err) => {
+            tracing::error!(
+                target: crate::INDEXER,
+                "Failed to save receipts and outcomes for shard {}: Error {}",
+                shard_id,
+                err
+            );
+
+            for (receipt, outcome) in receipts.into_iter().zip(outcomes) {
+                tx_collecting_storage
+                    .push_outcome_and_receipt_to_save(
```

Comment:
You should not use the remove_receipt_from_watching_list method, which is used to collect transactions. If you do, you may break the transaction collection process. You should pay attention to the push_outcome_and_receipt_to_save method. 
Consider separating the logic of adding receipts and outcomes.